dbclient/ClustersClient.py

```diff
@@ -277,6 +281,59 @@ def import_cluster_configs(self, log_file='clusters.log', acl_log_file='acl_clus
                 resp = self.put(api, acl_args)
                 print(resp)
 
+    def _log_cluster_ids_and_original_creators(
+            self,
+            cluster_log_file,
+            user_name_to_user_id_log_file='user_name_to_user_id.log',
+            creators_file='original_creator_user_ids.log',
+            cluster_ids_file='cluster_ids_to_change_creator.log'):
+        """
+        Log the cluster_ids_to_change_creator and original_creator_user_ids
+        These can be used to edit the clusters to be owned by the correct/original creator instead of the
+        PAT token owner. Fails if user_name_to_user_id.log does not exist.
+
+        :param cluster_log_file: file that contains the exported cluster objects.
+        :param creators_file: output file written with the list of original creators of clusters.
+                              The list should be in the same order as the cluster_ids_file.
+        :param cluster_ids_file: output file written with the list of cluster ids that need the creator tag to change.
+        """
+        cluster_ids_to_change_creator = []
+        original_creator_user_ids = []
+        with open(self.get_export_dir() + user_name_to_user_id_log_file, 'r') as fp:
+            user_name_to_user_id = json.loads(fp.read())
+
+        all_clusters = self.get_cluster_list(False)
+        # cluster_name -> (cluster_id, creator_user_name)
+        current_cluster_name_to_id_and_creator_name = {}
+        for cluster in all_clusters:
+            cluster_name = cluster['cluster_name']
+            if cluster_name in current_cluster_name_to_id_and_creator_name:
+                print(f"There are multiple clusters with name: {cluster_name}. Only the first cluster will have its "
+                      "creator changed.")
+            else:
+                current_cluster_name_to_id_and_creator_name[cluster_name] = \
+                    (cluster['cluster_id'], cluster['creator_user_name'])
+
+        with open(self.get_export_dir() + cluster_log_file, 'r') as fp:
+            for line in fp:
+                cluster_conf = json.loads(line)
+                cluster_name = cluster_conf['cluster_name']
+                original_cluster_creator = cluster_conf['creator_user_name']
+                current_cluster_id, current_cluster_creator = current_cluster_name_to_id_and_creator_name[cluster_name]
+
+                # Log only if the cluster's current creator is not the same as the original creator
+                if original_cluster_creator != current_cluster_creator:
+                    if cluster_name in current_cluster_name_to_id_and_creator_name and \
+                            original_cluster_creator in user_name_to_user_id:
+                        cluster_ids_to_change_creator.append(current_cluster_id)
+                        original_creator_user_ids.append(user_name_to_user_id[original_cluster_creator])
+
+        with open(self.get_export_dir() + cluster_ids_file, 'w') as fp:
```

Comment:
Just curious, no change required. 

Why do we output two separate files instead of writing related data into the same file? Copy-paste twice is more error-prone than once.

Reply:
It is because the Databricks side tool (the jenkins job for editing cluster owner) takes two separate parameters which are list. From the Jenkins job perspective, two-lists is the least error-prone way as other types of inputs are much easier to make mistakes on. (e.g. Json blobs, or list of tuples, etc) 

From this copy-pasting perspective, it is to reduce the chance of making mistakes on the Databricks side:
the invoker just gives these two files to Databricks, and Databricks directly copies pastes. Another similar way would be to make this in to one file, and let Databricks do the careful copy, but I think the current way is less error-prone.

Reply:
I agree, as long as we're providing the downloaded files directly, it should not be error prone. If we are copying / pasting values, that would be error prone and we shouldn't manually modify these values.

Reply:
yeah, whoever runs the Jenkins job needs to do the copying pasting, but no need to change values